# Understand Data
*final goal: common sensical/basic understanding of data variables, or what data you are given*

## References:
"In this dataset, you are given over a thousand low-dose CT images from high-risk patients in DICOM format. Each image contains a series with multiple axial slices of the chest cavity. Each image has a variable number of 2D slices, which can vary based on the machine taking the scan and patient."

"The DICOM files have a header that contains the necessary information about the patient id, as well as scan parameters such as the slice thickness."

"The images in this dataset come from many sources and will vary in quality. For example, older scans were imaged with less sophisticated equipment. You should expect the stage 2 data to be, on the whole, more recent and higher quality than the stage 1 data (generally having thinner slice thickness). Ideally, your algorithm should perform well across a range of image quality."

"Each patient id has an associated directory of DICOM files. The patient id is found in the DICOM header and is identical to the patient name. The exact number of images will differ from case to case, varying according in the number of slices. Images were compressed as .7z files due to the large size of the dataset.
* stage1.7z - contains all images for the first stage of the competition, including both the training and test set
* stage1_labels.csv - contains the cancer ground truth for the stage 1 training set images"

([kaggle-dsb data](https://www.kaggle.com/c/data-science-bowl-2017/data))

## Useful Kernels:
*these kernels help you understand the dataset better*
### [Anokas' Exploratory Data Analysis](https://www.kaggle.com/anokas/data-science-bowl-2017/exploratory-data-analysis)
shows you how to open DICOMs, and see the information in the headers which include:
  1. patient id
  2. scan parameters (z-index, slice thickness = pixel spacing, etc...)

### [Zuidhof's Full Preprocessing Tutorial](https://www.kaggle.com/gzuidhof/data-science-bowl-2017/full-preprocessing-tutorial)

## Opening Data Files

In [1]:
import pandas as pd

pd.read_csv('../../../../data/dsb/stage1_labels.csv')

id  cancer
0     0015ceb851d7251b8f399e39779d1e7d       1
1     0030a160d58723ff36d73f41b170ec21       0
2     003f41c78e6acfa92430a057ac0b306e       0
3     006b96310a37b36cccb2ab48d10b49a3       1
4     008464bb8521d09a42985dd8add3d0d2       1
5     0092c13f9e00a3717fdc940641f00015       0
6     00986bebc45e12038ef0ce3e9962b51a       0
7     00cba091fa4ad62cc3200a657aeb957e       0
8     00edff4f51a893d80dae2d42a7f45ad1       1
9     0121c2845f2b7df060945b072b2515d7       0
10    013395589c01aa01f8df81d80fb0e2b8       0
11    01de8323fa065a8963533c4a86f2f6c1       0
12    01e349d34c06410e1da273add27be25c       0
13    01f1140c8e951e2a921b61c9a7e782c2       0
14    024efb7a1e67dc820eb61cbdaa090166       0
15    0257df465d9e4150adef13303433ff1e       1
16    0268f3a7a17412178cfb039e71799a80       0
17    026be5d5e652b6a7488669d884ebe297       0
18    02801e3bbcc6966cb115a962012c35df       1
19    028996723faa7840bb57f57e28275e4c       1
20    0334c8242ce7ee1a6c1263096e4cc535       0
21    03fb0d0fdb187ee1160f09386b28c3f2       0
22    03ff23e445787886f8b0cb192b3c154d       0
23    043ed6cb6054cc13804a3dca342fa4d0       0
24    0482c444ac838adc5aa00d1064c976c1       0
25    04a3187ec2ed4198a25033071897bffc       0
26    04a52f49cdbfb8b99789b9e93f1ad319       0
27    04a8c47583142181728056310759dea1       1
28    04cfc5efa4c8c2a8944c8b9fa6cb04d1       0
29    04e5d435fa01b0958e3274be73312cac       0
...                                ...     ...
1367  fa45178d023325b255a3d4fc3e96cb7d       1
1368  fa744c0374ccc6aaf6711f463a9e5bc2       0
1369  fa7bf17071e3234ab1f350341303b174       0
1370  fa9575f64e6881c6b2730f0e225c9573       0
1371  fa968673987cfd078d91641294c3bf63       0
1372  fac65dbf7b6972049cfd37b5b122ec0b       0
1373  fb52dd8152e53a4ca7da5403d6d0db13       0
1374  fb57fc6377fd37bb5d42756c2736586c       1
1375  fb7ae70f05b6441ac4ea7187ce7c45f7       0
1376  fb7dfb6aae597d97c2da24179df0fe56       0
1377  fb99a80cbb2f441bb90135bab5b029fe       1
1378  fbaa8548e6c3753836579c7987d0034d       0
1379  fbae4d04285789dfa32124c86586dd09       1
1380  fbe0c3d6e4a50ca1c1bd3101515d0ab4       0
1381  fc545aa2f58509dc6d81ef02130b6906       1
1382  fc658f9bc12a751a17f16589f9e3619b       0
1383  fcdc2d423356d53759a8f602d40785a6       0
1384  fd0c2dfe0b0c58330675c3191cef0d5b       1
1385  fd2dd970bd3d91e5b26d7e57c03f70af       1
1386  fd4c2d4738bc25a5c331dbc101f3323a       0
1387  fd64b23b8cd8c371c8f76fbb503e4e0e       0
1388  fd7c0fb3c0e764357aa58e5f047be614       0
1389  fda187bfb1d6a2ecd4abd862c7f7f94c       1
1390  fdf2a2f5b86aea0da54732056fc7ab48       0
1391  fdf73dcce35167f3ed952a58f5a6f738       0
1392  fe26fd2bb25112b3c2ca93eb8e34f8ed       0
1393  fe45462987bacc32dbc7126119999392       1
1394  fe5c37e82b412833b8ad0abb57978377       0
1395  ff5d8e90500cf324e7b04a2f07cf0399       0
1396  ffe02fe7d2223743f7fb455dfaff3842       0

[1397 rows x 2 columns]

In [2]:
import pandas as pd

pd.read_csv('../../../../data/dsb/stage1_sample_submission.csv')

id  cancer
0    026470d51482c93efc18b9803159c960     0.5
1    031b7ec4fe96a3b035a8196264a8c8c3     0.5
2    03bd22ed5858039af223c04993e9eb22     0.5
3    06a90409e4fcea3e634748b967993531     0.5
4    07b1defcfae5873ee1f03c90255eb170     0.5
5    0b20184e0cd497028bdd155d9fb42dc9     0.5
6    12db1ea8336eafaf7f9e3eda2b4e4fef     0.5
7    159bc8821a2dc39a1e770cb3559e098d     0.5
8    174c5f7c33ca31443208ef873b9477e5     0.5
9    1753250dab5fc81bab8280df13309733     0.5
10   1cf8e778167d20bf769669b4be96592b     0.5
11   1e62be2c3b6430b78ce31a8f023531ac     0.5
12   1f6333bc3599f683403d6f0884aefe00     0.5
13   1fdbc07019192de4a114e090389c8330     0.5
14   2004b3f761c3f5dffb02204f1247b211     0.5
15   202898fa97c5949fbdc07ae7ff1cd9f0     0.5
16   21b73c938fd7d346ee77a60bd60aaeac     0.5
17   243038f7bb7787497c59bc17f04c6ed9     0.5
18   26142353f46d20c9fdded93f01e2bff4     0.5
19   263a1c3bfa43556623e75ed901e3fd8f     0.5
20   2703df8c469906a06a45c0d7ff501199     0.5
21   2a3e6ecf9499607ef4fd14b436136b0c     0.5
22   2d596b6ead89ab35577fe625a9a17cbb     0.5
23   2eb92d17ca91b393765e8acf069763a6     0.5
24   2f77fd993fbd858dec3c085b9ff1a3a2     0.5
25   3295cec04482210dc6f78c2b4a1d287b     0.5
26   33387bea2cacf6127035cc7033036a02     0.5
27   34037914ceeec5605fc890159dd425c5     0.5
28   38bf066bba822584e14c0af65d4bb5e9     0.5
29   3ee1fd6a0f3f108c3558e6699fb011f2     0.5
..                                ...     ...
168  d654966fd2498de023552b830c07a659     0.5
169  d753676c2c6c8ac6f97bd61ecab7554a     0.5
170  d81852bffda09dc8033a45332397c495     0.5
171  dbd9c8025907511e965e7abad955547d     0.5
172  e0aa61b44c33e6a75940a8541c6894c9     0.5
173  e314fd13809db0132443b924401d828b     0.5
174  e33c25d0dbca5e54385f2100ce523467     0.5
175  e3bc0a970a4af5d52826e06742f90e5b     0.5
176  e42065c1145ccf734312cb9edbe5234b     0.5
177  e60d99ea9648e1ce859eb0b386365e26     0.5
178  e6160ed0ff2eb214abd4df9a3c336c1d     0.5
179  e6d8ae8c3b0817df994a1ce3b37a7efb     0.5
180  e9a27e2645e1fad9434ce765f678585f     0.5
181  ea01deecde93cd9503a049d71d46e6d5     0.5
182  ea3a771ef05e288409e0250ea893cf87     0.5
183  eaeebb7a63edc8a329a7c5fbc583a507     0.5
184  eb9db3f740f8e153e85f83c57bc4e522     0.5
185  ebcdfabecf4b46b1e55e4a4c75a0afb0     0.5
186  efcb6def7a2080243052b6046186ab24     0.5
187  f0310ffc724faf9f7aef2c418127ee68     0.5
188  f4d23e0272a2ce5bfc7f07033d4f2e7d     0.5
189  f5ff7734997820b45dafa75dff60ece8     0.5
190  f7c387290d7e3074501eac167c849000     0.5
191  f89e3d0867e27be8e19d7ed50e1eb7e8     0.5
192  fad57a1078ddbc685e517bd8f24aa8ac     0.5
193  fb55849cee6473974612c17f094a38cd     0.5
194  fb5874408966d7c6bebd3d84a5599e20     0.5
195  fcfab3eddbdf0421c39f71d651cc5c56     0.5
196  fdcd385b0d2d12341661e1abe845be0b     0.5
197  ff8599dd7c1139be3bad5a0351ab749a     0.5

[198 rows x 2 columns]

## Summary:
### stage1.7z
* contains folders, each folder is a separate patient, 
* in each patient's folder are their CT scan image files
* CT scan image file
  * header will contain infomation (such as: patient id, scan thickness, etc...)
  * different number of image files for each person (based on type of equipment used)
  * many different image sources, and varying image quality (due to different equipment used)
  * each image for a patient is a 2D slice, combine all 2D slices (based on z-position) to get a 3D representation patient's lungs

### stage1_labels.csv
* contains the cancer ground truth labels for the stage 1 training set images (the correct answers for each patient id

### stage1_sample_submission.csv
* need to predict 198 for the test set

### Data Variables:
1. the images of the patient's lungs (2D versions compiled to make 3D representation)
2. scan thickness (or quality of the equipment)